docs/shim-rune.md

```diff
@@ -0,0 +1,241 @@
+encrypted unionfs (based sefs) image in the host.
+
+Since one image corresponds to a encrypted unionfs image, shim-rune can generate `container_id`
+based on image name and pass `container_id` to agent enclave contianer with `PullImageRequest`.
+The agent enclave contianer can store the encrypted unionfs image to directory distinguished by
+`container_id` in host, such as `<agent_container_bundle_path>/rootfs/images/<container_id>/sefs`.
+
+When shim-rune launch the app enclave container, shim-rune will find the corresponding encrypted
+unionfs image through `container_id`. Then shim-rune overlay the `encrypted unionfs image` and
+`boot instance bundle` to generate the final app enclave container bundle.
+
+### Gramine
+
+TODO
+
+At last, shim-rune call `rune` to launch the app enclave container. The process is similar to
+launch the agent enclave container with `rune`.
+
+## 4. Stop sandbox
+
+Compared with `containerd-shim-runc-v2`, shim-rune adds the work of kill/cleanup for the agent
+enclave container.
+
+# How to integrate with containerd
+
+Because shim-rune supports containerd shim v2 API, you can add the associated configurations for
+shim-rune in the containerd config file, e.g, `/etc/containerd/config.toml`, on your system.
+
+```
+        [plugins.cri.containerd]
+          ...
+          [plugins.cri.containerd.runtimes.rune]
+            runtime_type = "io.containerd.rune.v2"
+```
+then restart containerd on your system.
+
+# Appendix
+
+## shim configuration sample
+
+```
+log_level = "info" # "debug" "info" "warn" "error"
+
+[containerd]
+    socket = "/run/containerd/containerd.sock"
+    agent_container_instance = "/opt/enclave-cc/agent-instance/"
+    boot_container_instance = "/opt/enclave-cc/boot-instance/"
+    agent_container_root_dir = "/run/containerd/agentenclave"
+    agent_url = "tcp://0.0.0.0:7788"
+```
+
+where:
+- @log_level: specify the log level for shim-rune.
+- @socket: the containerd socket
+- @agent_container_instance:  the host path of pre-created oci bundle for agent enclave container
+- @boot_container_instance: the host path of pre-created oci bundle for boot instance
+- @agent_container_root_dir: the root dir of agent enclave container running state
+- @agent_url: the listening address of the agent container, through which the shim communicates
+with the agent container to perform PullImage. (support unix and tcp socket communication)
+  - Currently occlum 0.27 version does not support cross-world untrusted Unix domain socket.
+  - After occlum releases [NGO](https://github.com/occlum/ngo) at the end of June, the communication
+  based on tcp socket between shim-rune and agent container will be switched to ttrpc based on
+  cross-world untrusted Unix domain socket.
+
+# Reference
+
+- [kata container](https://github.com/confidential-containers/kata-containers-CCv0)
+- [contained](https://github.com/confidential-containers/containerd)
```

Comment:
```suggestion
- [containerd](https://github.com/confidential-containers/containerd)
```

Reply:
Thank you very much for your suggestions, I have modified the pr as you suggested. Thank you very much!